test/e2e/placement_selecting_resources_test.go

```diff
@@ -592,3 +592,190 @@ var _ = Describe("validating CRP when deleting resources in a matching namespace
+			Spec: placementv1beta1.ClusterResourcePlacementSpec{
+				ResourceSelectors: []placementv1beta1.ClusterResourceSelector{
+					{
+						Group:   "",
+						Kind:    "Namespace",
+						Version: "v1",
+						Name:    "invalid-namespace",
+					},
+				},
+				Strategy: placementv1beta1.RolloutStrategy{
+					RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+						UnavailablePeriodSeconds: pointer.Int(5),
+					},
+				},
+				RevisionHistoryLimit: pointer.Int32(1),
+			},
+		}
+		By(fmt.Sprintf("creating placement %s", crpName))
+		Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP %s", crpName)
+	})
+
+	AfterAll(func() {
+		By(fmt.Sprintf("deleting placement %s", crpName))
+		cleanupCRP(crpName)
+
+		By("deleting created work resources")
+		cleanupWorkResources()
+	})
+
+	It("should update CRP status as expected", func() {
+		crpStatusUpdatedActual := func() error {
+			return validateCRPStatus(types.NamespacedName{Name: crpName}, nil)
+		}
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP %s status as expected", crpName)
+	})
+
+	It("adding resource selectors", func() {
+		updateFunc := func() error {
+			crp := &placementv1beta1.ClusterResourcePlacement{}
+			if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+				return err
+			}
+
+			crp.Spec.ResourceSelectors = append(crp.Spec.ResourceSelectors, placementv1beta1.ClusterResourceSelector{
+				Group:   "",
+				Kind:    "Namespace",
+				Version: "v1",
+				Name:    fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess()),
+			})
+			// may hit 409
+			return hubClient.Update(ctx, crp)
+		}
+		Eventually(updateFunc(), eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update the crp %s", crpName)
+	})
+
+	It("should update CRP status as expected", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+	It("should have one policy snapshot revision and one resource snapshot revision", func() {
+		Expect(validateCRPSnapshotRevisions(crpName, 1, 1)).Should(Succeed(), "Failed to validate the revision history")
+	})
+
+	It("can delete the CRP", func() {
+		// Delete the CRP.
+		crp := &placementv1beta1.ClusterResourcePlacement{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+			},
+		}
+		Expect(hubClient.Delete(ctx, crp)).To(Succeed(), "Failed to delete CRP %s", crpName)
+	})
+
+	It("should remove placed resources from all member clusters", checkIfRemovedWorkResourcesFromAllMemberClusters)
+
+	It("should remove controller finalizers from CRP", func() {
+		finalizerRemovedActual := allFinalizersExceptForCustomDeletionBlockerRemovedFromCRPActual()
+		Eventually(finalizerRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove controller finalizers from CRP %s", crpName)
+	})
+})
+
+var _ = Describe("validating CRP revision history allowing multiple revisions when updating resource selector", Ordered, func() {
```

Comment:
I didn't get why this creates two resourceSnapshot while the one above creates one.  Is there any reason why this can not just be merged with the one above? They are like 95% the same.

Reply:
The above CRP sets the "RevisionHistoryLimit: pointer.Int32(1),".

They could be merged, but the whole process will be long as the CRP will be updated multiple times to update the revision limit & create resourcesnapshosts.

I feel the current approach is much cleaner.